examples/flower_classifier/image_pyfunc.py

```diff
@@ -0,0 +1,165 @@
+"""
+Example of a custom python function implementing image classifier with image preprocessing embedded
+in the model.
+"""
+import base64
+from io import BytesIO
+import keras
+import numpy as np
+import os
+import pandas as pd
+from PIL import Image
+import yaml
+import tensorflow as tf
+
+import mlflow
+import mlflow.keras
+from mlflow.utils.file_utils import TempDir
+from mlflow.utils.environment import _mlflow_conda_env
+
+
+def decode_and_resize_image(raw_bytes, size):
+    """
+    Read, decode and resize raw image bytes (e.g. raw content of a jpeg file).
+
+    :param raw_bytes: Image bits, e.g. jpeg image.
+    :param size: requested output dimensions
+    :return: Multidimensional numpy array representing the resized image.
+    """
+    return np.asarray(Image.open(BytesIO(raw_bytes)).resize(size), dtype=np.float32)
+
+
+class KerasImageClassifierPyfunc(object):
+    """
+    Image classification model with embedded pre-processing.
+
+    This class is essentially an MLflow custom python function wrapper around a Keras model.
+    The wrapper provides image preprocessing so that the model can be applied to images directly.
+    The input to the model is base64 encoded image binary data (e.g. contents of a jpeg file).
+    The output is the predicted class label, predicted class id followed by probabilities for each
+    class.
+
+    The model declares current local versions of Keras, Tensorlow and pillow as dependencies in its
+    conda environment file.
+    """
+
+    def __init__(self, graph, session, model, image_dims, domain):
+        self._graph = graph
+        self._session = session
+        self._model = model
+        self._image_dims = image_dims
+        self._domain = domain
+        probs_names = ["p({})".format(x) for x in domain]
+        self._column_names = ["predicted_label", "predicted_label_id"] + probs_names
+
+    def _predict_images(self, images):
```

Comment:
Maybe add a short doc comment to say what this is, or move it after predict where we saw how it's used